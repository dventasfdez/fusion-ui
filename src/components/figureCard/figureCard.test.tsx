import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import FigureCard, { FigureCardBody, FigureCardFigure, FigureCardFloatIcon, FigureCardHeader } from "./figureCard";

describe("Figure card snapshots", () => {
  it("Figure card", () => {
    const { container } = render(
      <FigureCard>
        <FigureCardFigure>00</FigureCardFigure>

        <FigureCardFloatIcon>
          <span>collections</span>
        </FigureCardFloatIcon>

        <FigureCardHeader>
          <h4>This is a title</h4>
        </FigureCardHeader>

        <FigureCardBody>This is a Figure in horizontal disposition</FigureCardBody>
      </FigureCard>
    );
    expect(container).toMatchSnapshot();
  });
});

describe("Figure card funcionality", () => {
  it("Figure card only with number", () => {
    const { getByTestId } = render(
      <FigureCard data-testid="figure-card">
        <FigureCardFigure>00</FigureCardFigure>
      </FigureCard>
    );
    const card = getByTestId("figure-card");
    expect(card).toHaveAttribute("style", "width: fit-content");
  });
  it("Figure card accent", () => {
    const { getByTestId } = render(
      <FigureCard accent data-testid="figure-card">
        <FigureCardFigure>00</FigureCardFigure>

        <FigureCardFloatIcon>
          <span>collections</span>
        </FigureCardFloatIcon>

        <FigureCardHeader>
          <h4>This is a title</h4>
        </FigureCardHeader>

        <FigureCardBody>This is a Figure in horizontal disposition</FigureCardBody>
      </FigureCard>
    );
    const card = getByTestId("figure-card");
    expect(card).toHaveClass("card_figure_accent");
  });
  it("Figure card selected", () => {
    const { getByTestId } = render(
      <FigureCard selected data-testid="figure-card">
        <FigureCardFigure>00</FigureCardFigure>

        <FigureCardFloatIcon>
          <span>collections</span>
        </FigureCardFloatIcon>

        <FigureCardHeader>
          <h4>This is a title</h4>
        </FigureCardHeader>

        <FigureCardBody>This is a Figure in horizontal disposition</FigureCardBody>
      </FigureCard>
    );
    const card = getByTestId("figure-card");
    expect(card).toHaveClass("card_figure_selected");
  });
  it("Figure card with onClick", () => {
    const onClick = jest.fn();
    const { getByTestId } = render(
      <FigureCard accent data-testid="figure-card" onClick={onClick}>
        <FigureCardFigure>00</FigureCardFigure>

        <FigureCardFloatIcon>
          <span>collections</span>
        </FigureCardFloatIcon>

        <FigureCardHeader>
          <h4>This is a title</h4>
        </FigureCardHeader>

        <FigureCardBody>This is a Figure in horizontal disposition</FigureCardBody>
      </FigureCard>
    );

    const card = getByTestId("figure-card");
    if (card) fireEvent.click(card);
    expect(onClick).toHaveBeenCalled();
  });
});
