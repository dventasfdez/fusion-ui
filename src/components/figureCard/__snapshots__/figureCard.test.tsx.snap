// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figure card funcionality Figure card only with number 1`] = `
<div
  class="card_figure "
  data-testid="figure-card"
>
  <div
    class="card-figure "
  >
    00
  </div>
</div>
`;

exports[`Figure card snapshots Figure card 1`] = `
<div>
  <div
    class="card_figure "
  >
    <div
      class="card-figure "
    >
      <span
        class="card-float-icon  "
      >
        collections
      </span>
      00
    </div>
    <div
      class="card-header "
    >
      <h4>
        This is a title
      </h4>
    </div>
    <div
      class="card-body "
    >
      This is a Figure in horizontal disposition
    </div>
  </div>
</div>
`;
